src/OSPSuite.Presentation/Presenters/Charts/ChartDisplayPresenter.cs

```diff
@@ -580,30 +573,23 @@ private bool canHandle(ChartEvent chartEvent)
          return Equals(Chart, chartEvent.Chart);
       }
 
-      public void ShowContextMenu(Curve curve, Point popupLocation)
+      public void ActivateFirstContextMenuEntryFor(Axis axis)
       {
-         _curveContextMenuFactory.CreateFor(curve, this).Show(_view, popupLocation);
+         _contextMenuFactory.CreateFor(viewItemFor(axis), this).ActivateFirstMenu();
       }
 
-      public void ShowContextMenu(Axis axis, Point popupLocation)
+      public void ActivateFirstContextMenuEntryFor(Curve curve)
       {
-         _axisContextMenuFactory.CreateFor(axis, this).Show(_view, popupLocation);
+         _contextMenuFactory.CreateFor(viewItemFor(curve), this).ActivateFirstMenu();
       }
 
-      public void ShowContextMenu(CurveChart curveChart, Point popupLocation)
+      public void ShowContextMenu(IViewItem viewItem, Point popupLocation)
       {
-         var contextMenu = _curveChartContextMenuFactory.CreateFor(curveChart, this);
```

Comment:
The whole context menu implementation was sub-optimal here. We were using one explicit factory per type whereas the presenter should only use one and delegate to the factory the creation of the context menu. This will make this presenter a bit lighter (in terms of dependencies that is)